boa3_test/tests/compiler_tests/test_for.py

```diff
@@ -1,16 +1,13 @@
-from boa3_test.tests.boa_test import BoaTest  # needs to be the first import to avoid circular imports
-
 from boa3.internal.exception import CompilerError
 from boa3.internal.neo.vm.opcode.Opcode import Opcode
 from boa3.internal.neo.vm.type.Integer import Integer
-from boa3.internal.neo3.vm import VMState
-from boa3_test.tests.test_drive.testrunner.boa_test_runner import BoaTestRunner
+from boa3_test.tests import boatestcase
 
 
-class TestFor(BoaTest):
+class TestFor(boatestcase.BoaTestCase):
     default_folder: str = 'test_sc/for_test'
 
-    def test_for_tuple_condition(self):
+    async def test_for_tuple_condition(self):
```

Comment:
Can you break the tests that test both the compilation and execution in two different tests?
This one for example, into `test_for_tuple_condition_compile` and `test_for_tuple_condition_run`